temba/campaigns/views.py

```diff
@@ -236,6 +246,76 @@ def save(self, obj):
+            SELECT row_to_json(flow_data)
+            FROM (
+              SELECT min(fs.created_on)                                       as start_time,
+                     max(fs.modified_on)                                      as updated_time,
+                     sum(fs.contact_count)                                    as total_contacts,
+                     count(ct.id) filter ( where ct.is_active = false )       as invalid_contacts,
+                     count(fr.id) filter ( where fr.status not in ('A', 'W')) as reached_contacts,
+                     count(fr.id) filter ( where fr.responded = true )        as bounces,
+                     max(case when fr.status in ('S', 'P') then 1 else 0 end) as has_running,
+                     count(ct.id) filter ( where ct.status = 'S' )            as opt_outs,
+                     count(fr.id) filter ( where fr.status = 'F')             as carrier_errors,
+                     count(fr.id) filter ( where fr.status in ('E', 'I'))     as ccl_errors,
+                     sum((
+                       SELECT count(*) filter ( where evt->>'type' = 'msg_received' )
+                       FROM jsonb_array_elements(fr.events) evt)
+                     ) as inbound
+              FROM flows_flowstart as fs
+              LEFT JOIN flows_flowrun as fr on fs.id = fr.start_id
+              LEFT JOIN contacts_contact as ct on fr.contact_id = ct.id
+              WHERE fs.flow_id = ce.flow_id
+              GROUP BY fs.flow_id
+            ) flow_data) flow_data
+          FROM campaigns_campaign as cp LEFT JOIN campaigns_campaignevent as ce on ce.campaign_id = cp.id
+          WHERE cp.id = 2 AND ce.event_type = 'F'
+          GROUP BY cp.id, ce.flow_id
+        ) t GROUP BY t.id;
+        """
+
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+
+            campaign = self.get_object()
+            data = Campaign.objects.raw(self.select_data_sql, params=[campaign.id, campaign.id])
+            context["current_time"] = timezone.now()
+            try:
+                context["start_time"] = data[0].start_time
+                context["updated_time"] = data[0].updated_time
+                context["end_time"] = data[0].updated_time if not data[0].has_running else None
+                context["total_contacts"] = data[0].total_contacts
+                context["invalid_contacts"] = data[0].invalid_contacts
+                context["reached_contacts"] = data[0].reached_contacts
+                context["remaining_contacts"] = data[0].total_contacts - data[0].reached_contacts
+                context["bounces"] = data[0].bounces
+                context["inbound"] = data[0].inbound
+                context["opt_outs"] = data[0].opt_outs
+                context["ccl_errors"] = data[0].ccl_errors
+                context["carrier_errors"] = data[0].carrier_errors
+            except IndexError:
+                pass
```

Comment:
Does `IndexError` mean that `data` is empty?

Reply:
Yes